lib/uniffi-zcash/src/zcash_primitives/transaction/mod.rs

```diff
@@ -3,3 +3,241 @@ pub use self::components::*;
 
 mod fees;
 pub use self::fees::*;
+
+use std::sync::{Arc, RwLock};
+
+use zcash_primitives::transaction::{
+    builder::Builder,
+    components::{sapling::builder::SaplingMetadata, Amount},
+    Transaction,
+};
+
+use crate::{
+    SecpSecretKey, ZcashBlockHeight, ZcashConsensusParameters, ZcashDiversifier, ZcashError,
+    ZcashExtendedSpendingKey, ZcashLocalTxProver, ZcashMemoBytes, ZcashOutgoingViewingKey,
+    ZcashPaymentAddress, ZcashResult, ZcashSaplingMerklePath, ZcashSaplingNote,
+    ZcashTransparentAddress,
+};
+
+pub use self::components::*;
+
+pub struct ZcashTransactionBuilder {
```

Comment:
`ZcashTransactionBuilder` is a simple, UniFFI compatible builder type, which stores the user inputs in `Vec`s and defers the [original builder](https://docs.rs/zcash_primitives/0.10.2/zcash_primitives/transaction/builder/struct.Builder.html) instantiation/execution to the `build()` function (see below). This is mainly done in this way because the original builder type has generic type parameters:

```rust
/// Generates a [`Transaction`] from its inputs and outputs.
pub struct Builder<'a, P, R> {
    params: P,
    rng: R,
    // ...
}
```

Please feel free to provide other alternatives here.